pdq/DMS.go

```diff
@@ -0,0 +1,18 @@
+package pdq
+
+import (
+	"errors"
+	"math"
+)
+
+func CalcDMS(abstractness float64, instability float64) (float64, error) {
+	if abstractness < 0 {
+		return -1, errors.New("abstractness cannot be lesser than 0")
+	}
+
+	if instability < 0 {
+		return -1, errors.New("instability cannot be lesser than 0")
+	}
```

Comment:
if the logic for both checks is the same, its okay to combine them to avoid code duplication

like
abstractness < 0 || instability < 0

also
think its better to provide more context in the error messages, you can use formatted error strings that include the values that caused the error

smth like
_invalid metric value: abstractness %.2f, instability %.2f; values must not be less than 0_